MovieApp/MovieApp/MoviesViewController.m

```diff
@@ -19,69 +26,76 @@ @implementation MoviesViewController
 - (void)viewDidLoad {
     
     [super viewDidLoad];
+    [self configureView];
+
+    [self.sortSegmentedControl setSelectedSegmentIndex:2];
+    [self.sortSegmentedControl sendActionsForControlEvents:UIControlEventValueChanged];
+}
+
+-(void)configureView{
     
-     [_moviesCollectionView registerNib:[UINib nibWithNibName:@"MoviesCollectionViewCell" bundle:nil]  forCellWithReuseIdentifier:@"collectionCell"];
-    
-    searchBar = [[UISearchBar alloc] init];
-    leftButton = [[UIBarButtonItem alloc]init];
-    rightButton = [[UIBarButtonItem alloc]init];
-    
+    [_moviesCollectionView registerNib:[UINib nibWithNibName:[MovieAppConfiguration getMoviesCollectionViewCellNibName] bundle:nil]  forCellWithReuseIdentifier:[MovieAppConfiguration getMoviesCollectionViewCellIdentifier]];
+    searchBar=[[UISearchBar alloc]init];
     searchBar.placeholder=@"Search";
     
-    leftButton.title=@"left";
-    leftButton.tintColor=[UIColor whiteColor];
-    
-    rightButton.title=@"right";
-    rightButton.tintColor=[UIColor whiteColor];
+    UITextField *txfSearchField = [searchBar valueForKey:TEXT_FIELD_PROPERTY_NAME];
+    txfSearchField.backgroundColor = [UIColor darkGrayColor];
     
     self.navigationItem.titleView = searchBar;
-    self.navigationItem.leftBarButtonItem = leftButton;
-    self.navigationItem.rightBarButtonItem = rightButton;
-    
-    MovieDBDownloader *downloader = [MovieDBDownloader alloc];
-    [downloader configure];
-    [downloader getdMoviesByCriterion:TOP_RATED returnToHandler:self];
-    
 }
 
-
 - (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section{
-    return [movies count];
+    return movies ? [movies count] : 0;
     
 }
 
 - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
     
-     MoviesCollectionViewCell *cell = [_moviesCollectionView dequeueReusableCellWithReuseIdentifier:[MoviesCollectionViewCell cellIdentifier] forIndexPath:indexPath];
+    MoviesCollectionViewCell *cell = [_moviesCollectionView dequeueReusableCellWithReuseIdentifier:[MoviesCollectionViewCell cellIdentifier] forIndexPath:indexPath];
     
-    cell.titleLabel.text=((Movie *)movies[indexPath.row]).title;
+    Movie *currentMovie=(Movie *)movies[indexPath.row];
     
-    return cell;
+    cell.titleLabel.text=currentMovie.title;
+    
+    NSDateFormatter *dateFormatter=[[NSDateFormatter alloc]init];
+    [dateFormatter setDateFormat:PREFFERED_DATE_FORMAT];
+    cell.releaseDateLabel.text=[dateFormatter stringFromDate:currentMovie.releaseDate];
+    
+    //hardcoded for now
+    cell.genreLabel.text = @"Mistery, Thriller";
+    cell.ratingLabel.text=[NSString stringWithFormat:@"%.1f", currentMovie.voteAverage];
```

Comment:
Please, add setup method for cell configuration to MoviesCollectionViewCell, like you did with news feed cell. This is practice that should be consistent!